contracts/Market.sol

```diff
@@ -651,28 +611,30 @@ contract Market is PausableAccessPreset {
   }
 
   /**
-   * @notice Removes the removal from active supply and inserts it into the reserved supply,
-   * where it cannot be used to fill orders.
-   *
-   * @dev If the removal is the last for the supplier, removes the supplier from the active supplier queue.
-   *
-   * todo reserveRemoval can be generalized as into withdrawRemoval so we don't have to track reserved supply
+   * todo
    */
-  function reserveRemoval(uint256 removalId)
-    external
-    whenNotPaused
-    onlyRole(RESERVER_ROLE)
-  {
-    address supplierAddress = RemovalIdLib.supplierAddress(removalId);
-    _removeActiveRemoval(supplierAddress, removalId);
-    if (!_reservedSupply.add(removalId)) {
-      revert RemovalAlreadyReserved({removalId: removalId});
+  function withdraw(uint256 removalId) external whenNotPaused {
+    if (
+      _msgSender() == RemovalIdLib.supplierAddress(removalId) ||
+      hasRole(DEFAULT_ADMIN_ROLE, _msgSender())
+    ) {
+      address supplierAddress = RemovalIdLib.supplierAddress(removalId);
```

Comment:
since the supplier address is also used on line 618, maybe do the lookup once at the top of the function?